core/modules/loader/LoaderMsg.h

```diff
@@ -0,0 +1,367 @@
+
+    UInt32Element() : NumElement(MYTYPE) {}
+    explicit UInt32Element(uint32_t element_) : NumElement(MYTYPE, element_) {}
+    UInt32Element(UInt32Element const&) = delete;
+    UInt32Element& operator=(UInt32Element const&) = delete;
+};
+
+
+class UInt64Element : public NumElement<uint64_t> {
+public:
+    using Ptr = std::shared_ptr<UInt64Element>;
+    static const int MYTYPE = UINT64_ELEM;
+
+    UInt64Element() : NumElement(MYTYPE) {}
+    explicit UInt64Element(uint64_t element_) : NumElement(MYTYPE, element_) {}
+    UInt64Element(UInt64Element const&) = delete;
+    UInt64Element& operator=(UInt64Element const&) = delete;
+};
+
+
+
+// Using uint16_t for length in all cases.
+class StringElement : public MsgElement {
+public:
+    using Ptr = std::shared_ptr<StringElement>;
+    using UPtr = std::unique_ptr<StringElement>;
+    static const int myType = STRING_ELEM;
+
+    StringElement(std::string const& element_) : MsgElement(myType), element(element_) {}
+    StringElement() : MsgElement(myType) {}
+    StringElement(StringElement const&) = delete;
+    StringElement& operator=(StringElement const&) = delete;
+
+    ~StringElement() override = default;
+
+    bool appendToData(BufferUdp& data) override;
+    bool retrieveFromData(BufferUdp& data) override;
+    std::string getStringVal() override { return element; }
+
+    std::string element;
+
+    bool equal(MsgElement* other) override {
+        StringElement* ptr = dynamic_cast<StringElement*>(other);
+        if (ptr == nullptr) { return false; }
+        return (ptr->element == ptr->element);
+    }
+
+    /// The size of StringElement changes!
+    size_t transmitSize() const override {
+        // char in string, variable to transmit string length, size of base class.
+        return element.size()+ sizeof(uint16_t) + sizeOfBase();
+    }
+
+    template<typename T>
+    std::unique_ptr<T> protoParse() {
+        std::unique_ptr<T> protoItem(new T());
+        bool success = proto::ProtoImporter<T>::setMsgFrom(*protoItem, element.data(), element.length());
+        if (not success) {
+            return nullptr;
+        }
+        return protoItem;
+    }
+
+
+    /// This function is only usable if it is know that all data has been read from the socket.
+    /// This the case with UDP, and boost asio async reads that return after X bytes read.
+    template<typename T>
+    static std::unique_ptr<T> protoParse(BufferUdp& data) {
+        StringElement::Ptr itemData = std::dynamic_pointer_cast<StringElement>(MsgElement::retrieve(data));
+        if (itemData == nullptr) { return nullptr; }
+        return itemData->protoParse<T>();
+    }
+};
+
+
+/// Base class for loader messages.
+//  These messages are meant to be short and simple UDP messages. Long messages
+//  may have difficulty being transmitted successfully.
+//
+//  The message contains the message kind and the address of the entity sending
+//  the message.
+//
+class LoaderMsg {
+public:
+    enum {
```

Comment:
What does it define? States of in the communication FSA? Or just a bag of constance which would be a VERY BAD idea :) In that (the later) case I would recommend splitting that "bag" into a few named `enum` types grouping constants accordingly.

Also, why it's anonymous?

I always find it useful to complement these `enum` types with a static function:
```
enum States {
  IDLE = 0,
  IN_PROGRESS,
  ...
};
static std::string state2str(State state);
```
This is very handy if you need to dump enums into the log streams.